mlir/test/Dialect/Tosa/constant-matmul.mlir

```diff
@@ -0,0 +1,41 @@
+// RUN: mlir-opt --split-input-file --tosa-layerwise-constant-fold %s | FileCheck %s
+
```

Comment:
Can you add tests where no folding happens (e.g. because of the element type)?
What would happen for dynamic shapes (`?`) and unranked tensors (`*`)? Can you add tests for those as well?

Reply:
added test for them.